subsys/bluetooth/mesh/main.c

```diff
@@ -105,6 +105,8 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 	}
 
 	err = bt_mesh_net_create(net_idx, flags, net_key, iv_index);
+	bt_mesh_net_settings_commit();
```

Comment:
Shouldn't it be part of `bt_mesh_start` procedure? I guess it is better to run some internal timers when the device passed all conditional start functions.

Probably somewhere here: 
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/main.c#L426

Reply:
The stack should not start this timer until the device is actually provisioned. The bt_mesh_start() on the other hand can technically be called any time as such.

Reply:
Indeed, for example, the next power on cycle it will be called from here: https://github.com/alxelax/zephyr/blob/main/subsys/bluetooth/mesh/settings.c#L90
and will start IV update timer.